src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -4232,6 +4276,21 @@ class TableExpressionSegment(ansi.TableExpressionSegment):
     )
 
 
+class PartitionBySegment(BaseSegment):
+    """A `PARTITION BY` for `copy_into_location` functions."""
+
+    type = "partition_by_clause"
```

Comment:
This already exists in BigQuery but with a different type. Let's keep them consistent:

```suggestion
    type = "partition_by_segment"
```

Though we probably should be consistent with `PartitionClauseSegment` from ANSI too. But let's leave that for another day and at least not diverge to a third one.

Reply:
Change pushed